spring-cloud-sleuth-otel-autoconfigure/src/main/java/org/springframework/cloud/sleuth/autoconfig/otel/OtelResourceProperties.java

```diff
@@ -34,6 +37,11 @@ public class OtelResourceProperties {
 	 */
 	private boolean enabled = true;
 
+	/**
+	 * Map of custom resource attributes (e.g. service.version)
+	 */
+	private Map<String, String> attributes = null;
```

Comment:
we can always initialize this to an empty map - that way no NPE checks have to take place